app/src/main/java/com/vultisig/wallet/ui/screens/keygen/GeneratingKey.kt

```diff
@@ -65,10 +64,7 @@ internal fun GeneratingKey(
 
         KeygenState.Success -> {
             LaunchedEffect(Unit) {
-                viewModel.saveVault()
-                Thread.sleep(2000) // wait for 2 seconds
-                viewModel.stopService(context)
-                navController.navigate(Screen.Home.route)
+                viewModel.saveVault(context)
```

Comment:
Can you please make sure we wait for 2 seconds before we shutdown the mediator service?  This give the pair device a chance to get the last messages , or send message across.

Reply:
done